File: backend/src/SiteYonetimi.Infrastructure/Entities/Banka.cs
using SiteYonetimi.Shared.Entities;

namespace SiteYonetimi.Infrastructure.Entities;

public class Banka : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;

    public ICollection<BankaSubesi> Subeler { get; set; } = new List<BankaSubesi>();
}

File: frontend/src/types/banka.ts
export interface BankaSubesi {
  id: string
  bankaId: string
  subeAdi: string
  subeKodu?: string | null
  isActive: boolean
}

export interface Banka {
  id: string
  name: string
  isActive: boolean
  subeler: BankaSubesi[]
}

export interface BankaSubesiPickerItem {
  id: string
  bankaAdi: string
  subeAdi: string
  subeKodu?: string | null
}

export interface CreateBankaDto { name: string }
export interface UpdateBankaDto { name: string; isActive: boolean }
export interface CreateBankaSubesiDto { subeAdi: string; subeKodu?: string }
export interface UpdateBankaSubesiDto { subeAdi: string; subeKodu?: string; isActive: boolean }
